src/App.js

```diff
@@ -2,38 +2,42 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-const useSemiPersistentState = ()=>{
 
- const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
 
+function App() {
 
 
-   useEffect(()=>{
 
-     localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-     JSON.parse(localStorage.getItem('savedTodoList'));
-     
-     },[todoList]);
+const [todoList, setTodoList] = useState([]);
 
-     return [todoList,setTodoList]
-};
+const [isLoading, setIsLoading] = useState(true);
 
-function App() {
 
+  useEffect(()=>{
+     new Promise((resolve, reject)=>{
+       setTimeout(
+      () => resolve({ data: { todoList:  JSON.parse(localStorage.getItem('savedTodoList')) || [] } }),
+      2000
+    )
+     }).then((result)=>{
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+     });
+    
+  },[]);
 
-const [todoList,setTodoList] = useSemiPersistentState();
 
-const removeTodo = (id)=>{
-     function removeTodo(id) {
+   useEffect(()=>{
+    if (!isLoading){
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+     JSON.parse(localStorage.getItem('savedTodoList'));
+    };
+     },[todoList]);
```

Comment:
Couple of things:
- make sure that you put `isLoading` in the dependency array. You will want `useEffect` to run when `isLoading` changes. 
- you do not need `JSON.parse(localStorage.getItem('savedTodoList'));` in this function. Your first useEffect will take care of getting the item from local storage.

```suggestion
       useEffect(() => {
    if (!isLoading) {
      localStorage.setItem('savedTodoList', JSON.stringify(todoList));
    }
  }, [todoList, isLoading]);
```